core/src/com/nuclearthinking/myheroagency/ui/font/FontFactory.java

```diff
@@ -1,35 +1,56 @@
 package com.nuclearthinking.myheroagency.ui.font;
 
-import com.badlogic.gdx.Gdx;
 import com.badlogic.gdx.graphics.g2d.BitmapFont;
 import com.badlogic.gdx.graphics.g2d.freetype.FreeTypeFontGenerator;
-import com.nuclearthinking.myheroagency.utils.Constants;
-
-/**
- * Date: 08.05.2016
- * Time: 11:22
- *
- * @author Vladislav Radchenko (onfient@gmail.com)
- */
+import com.nuclearthinking.myheroagency.controller.Asset;
+
+import java.util.HashMap;
+
+import static com.nuclearthinking.myheroagency.utils.Constants.ALL_CHARACTERS;
+
 public class FontFactory {
+    private static final HashMap<Integer, BitmapFont> robotoLightMap = new HashMap<Integer, BitmapFont>();
+    private static final HashMap<Integer, BitmapFont> robotoBoldMap = new HashMap<Integer, BitmapFont>();
 
-    private static final FreeTypeFontGenerator generator = new FreeTypeFontGenerator(Gdx.files.internal("font/Kramola.ttf"));
-    private static BitmapFont font;
+    private FreeTypeFontGenerator robotoLightGenerator;
+    private FreeTypeFontGenerator robotoBoldGenerator;
+
+    public BitmapFont getRobotoLight(int size) {
+        BitmapFont font = robotoLightMap.get(size);
+        if (font == null) {
+            FreeTypeFontGenerator.FreeTypeFontParameter parameter = new FreeTypeFontGenerator.FreeTypeFontParameter();
+            parameter.characters = ALL_CHARACTERS;
+            parameter.size = size;
+            font = getRobotoLightGenerator().generateFont(parameter);
+            robotoLightMap.put(size, font);
+        }
+        return font;
+    }
 
-    public static BitmapFont getFont() {
+    public BitmapFont getRobotoBold(int size) {
+        BitmapFont font = robotoBoldMap.get(size);
         if (font == null) {
             FreeTypeFontGenerator.FreeTypeFontParameter parameter = new FreeTypeFontGenerator.FreeTypeFontParameter();
-            parameter.characters = Constants.ALL_CHARACTERS;
-            parameter.size = 21;
-            generateFont(parameter);
+            parameter.characters = ALL_CHARACTERS;
+            parameter.size = size;
+            font = getRobotoBoldGenerator().generateFont(parameter);
+            robotoBoldMap.put(size, font);
```

Comment:
не выйдет, я пробовал, потом при пересоздании генератора  после диспоуза
падает нативная либа

да и не к чему диспоузить если он реиспользуется и привязан только к одному
ресурсу

вс, 15 мая 2016 г., 16:00 Izonami <notifications@github.com>:

> In core/src/com/nuclearthinking/myheroagency/ui/font/FontFactory.java
> <https://github.com/nuclearthinking/myheroagency/pull/29#discussion_r63291270>
> :
>
> >          if (font == null) {
> >              FreeTypeFontGenerator.FreeTypeFontParameter parameter = new FreeTypeFontGenerator.FreeTypeFontParameter();
> > -            parameter.characters = Constants.ALL_CHARACTERS;
> > -            parameter.size = 21;
> > -            generateFont(parameter);
> > +            parameter.characters = ALL_CHARACTERS;
> > +            parameter.size = size;
> > +            font = getRobotoBoldGenerator().generateFont(parameter);
> > +            robotoBoldMap.put(size, font);
>
> хорошо бы наверное диспоузить генератор после присвоение к битмапфонту
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/nuclearthinking/myheroagency/pull/29/files/1f35520af09839113aaa048535852188aac8d92b#r63291270>
>